ks_patches/vaksine_utils.js

```diff
@@ -0,0 +1,170 @@
+import {PROFIL_VAKSINE_1_TYPE_ID, PROFIL_VAKSINE_1_DATO_ID, PROFIL_VAKSINE_2_TYPE_ID, PROFIL_VAKSINE_2_DATO_ID, PROFIL_VAKSINE_3_TYPE_ID, PROFIL_VAKSINE_3_DATO_ID} from "../utils/constants";
+
+export function createCombinedVaccineObject(sysvakVaccine, selectedTei, DateUtils) {
+    return sysvakVaccine.map((vaccine, index) => createVaccineObject(selectedTei, vaccine, index + 1, DateUtils));
+}
+
+function createVaccineObject(selectedTei, vaccine, doseNr, DateUtils) {
+    var vaccinationDate = convertVaccineDate(vaccine.vaccinationDate, DateUtils);
+    return enrichWithValidation({
+        profileDose: getVaksineDoseFromProfile(selectedTei, doseNr),
+        ...getVaccineNicenameAndType(vaccine),
+        date: vaccinationDate,
+        monthsAgo: getNrOfMonthsAgo(vaccinationDate, DateUtils)
+    });
+}
+
+function getNrOfMonthsAgo(date, DateUtils) {
+    if (!date) {
+        return undefined;
+    }
+    var age = DateUtils.getAge(date);
+    if(age.years > 0) {
+        return 12;
+    }
+    return age.months;
+}
+
+function getVaksineDoseFromProfile(selectedTei, doseNr) {
+    if(doseNr > 3 || doseNr < 1) {
+        return undefined;
+    }
+    var ids = getVaksineIds(doseNr);
+    return getNameAndDate(ids, selectedTei.attributes);
+}
+
+function getVaksineIds(doseNr) {
+    if(doseNr === 1) {
+        return {
+            typeId: PROFIL_VAKSINE_1_TYPE_ID,
+            dateId: PROFIL_VAKSINE_1_DATO_ID
+        }
+    }
+    if(doseNr === 2) {
+        return {
+            typeId: PROFIL_VAKSINE_2_TYPE_ID,
+            dateId: PROFIL_VAKSINE_2_DATO_ID
+        }
+    }
+    if(doseNr === 3) {
+        return {
+            typeId: PROFIL_VAKSINE_3_TYPE_ID,
+            dateId: PROFIL_VAKSINE_3_DATO_ID
+        }
+    }
+}
+
+function getNameAndDate(ids, attributes) {
+    var dose = {
+        name: attributes.find(att => att.attribute === ids.typeId).value,
+        date: attributes.find(att => att.attribute === ids.dateId).value
+    };
+    return dose && dose.date || dose.name ? dose : undefined;
+}
+
+function convertVaccineDate(vaccinationDate, DateUtils) {
+    const tmpDate = angular.copy(vaccinationDate);
+    if(tmpDate[1]) {
+        tmpDate[1] = tmpDate[1] - 1;
+    }
+    return DateUtils.getDateFromUTCString(tmpDate);
+}
+
+function getVaccineNicenameAndType(vaccine) {
+    switch(vaccine.vaccineCode.code) {
```

Comment:
Is these also mapped to an option set where you could use the name from - instead of hardcoding here?

Reply:
I got them from an email from FHI, it is also a subset from a PDF of vaccine codes. Not used anywhere else.